lib/utils.js

```diff
@@ -441,6 +441,56 @@ const executeOperation = (topology, operation, args, options) => {
   });
 };
 
+/**
+ * Applies a write concern to a command based on well defined inheritance rules, optionally
+ * detecting support for the write concern in the first place.
+ *
+ * @param {Object} target the target command we will be applying the write concern to
+ * @param {Object} sources sources where we can inherit default write concerns from
+ * @param {Object} [options] optional settings passed into a command for write concern overrides
+ * @returns {Object} the (now) decorated target
+ */
+function applyWriteConcern(target, sources, options) {
+  options = options || {};
+  const db = sources.db;
+  const coll = sources.collection;
+  const topology = db ? db.s.topology : coll.s.topology;
+
+  if (!topology.capabilities().commandsTakeWriteConcern) {
+    return target;
+  }
+
+  // NOTE: there is probably a much better place for this
+  if (db && db.s.options.retryWrites) {
+    target.retryWrites = true;
+  }
+
+  if (options.w != null || options.j != null || options.fsync != null) {
```

Comment:
Under this behavior, if the user overrides either `w`, `j`, or `fsync`, we will ignore the entire default write concern.

For example if the collection has a `writeConcern` with `w` and `j` defined, and the user specifies a new value for `w`, our new `writeConcern` will only have `w`, not `j`.

Is that expected behavior?

Also, should the user be able to specify a new writeConcern value for `wtimeout` only?